src/Model/ChangeStreamIterator.php

```diff
@@ -140,10 +150,10 @@ public function current()
      * but it's very much an invalid use-case. This method can be dropped in 2.0
      * once the class is final.
      */
-    final public function getInnerIterator(): Cursor
+    final public function getInnerIterator(): CursorInterface
```

Comment:
Since the constructor was changed to require a `CursorInterface<int, TValue>&Iterator<int, TValue>` should we use Psalm to document that return type as well? I understand we can't use actual PHP syntax to do so yet.

Likewise, does it make sense to assert that the inner iterator is _both_ CursorInterface and Iterator below?

Reply:
I've updated the return type once again to be `Iterator` (as that's what the parent class requires) and added a `@return` annotation to annotate the full type. I've also added assertions for both `CursorInterface` and `Iterator` in the method body.